src/toil/jobStores/aws/jobStore.py

```diff
@@ -681,6 +687,13 @@ def bucket_creation_pending(e):
                             raise
                         else:
                             return None
+                    elif e.status == 301:
+                        # This is raised if the user attempts to get a bucket in a region outside
+                        # the specified one, if the specified one is not `us-east-1`.  The us-east-1
+                        # server allows a user to use buckets from any region.
+                        assert self.region != 'us-east-1'
```

Comment:
I wouldn't assert this. Assertions are for things that you could break by changing your code. This assertion depends on Amazon. If Amazon decides to send redirects from us-east-1 this would break.